fastlane_bot/tests/deterministic/dtest_tx_helper.py

```diff
@@ -0,0 +1,290 @@
+                transactions.append(file.read())
+        return transactions
+
+    def tx_scanner(self, args: argparse.Namespace) -> list:
+        """
+        Scan for successful transactions in the most recent log folder.
+        
+        Args:
+            args (argparse.Namespace): The command-line arguments.
+            
+        Returns:
+            list: A list of successful transactions.
+        """
+        most_recent_log_folder = self.find_most_recent_log_folder()
+        args.logger.debug(f"Accessing log folder {most_recent_log_folder}")
+
+        pool_data_file = os.path.join(most_recent_log_folder, "latest_pool_data.json")
+        if self.wait_for_file(pool_data_file, args.logger):
+            pool_data = self.load_json_data(pool_data_file)
+            args.logger.debug(f"len(pool_data): {len(pool_data)}")
+
+        transactions = self.read_transaction_files(most_recent_log_folder)
+        successful_txs = [tx for tx in transactions if "'status': 1" in tx]
+        args.logger.debug(f"Found {len(successful_txs)} successful transactions.")
+
+        return successful_txs
+
+    @staticmethod
+    def clean_tx_data(tx_data: dict) -> dict:
+        """
+        This method takes a transaction data dictionary and removes the cid0 key from the trades.
+        
+        Args:
+            tx_data (dict): The transaction data.
+            
+        Returns:
+            dict: The cleaned transaction data.
+        """
+        if not tx_data:
+            return tx_data
+
+        for trade in tx_data["trades"]:
+            if trade["exchange"] == "carbon_v1" and "cid0" in trade:
+                del trade["cid0"]
+        return tx_data
+
+    @staticmethod
+    def get_tx_data(strategy_id: int, txt_all_successful_txs: list) -> dict:
+        """
+        This method takes a list of successful transactions and a strategy_id and returns the transaction data for the
+        given strategy_id.
+        
+        Args:
+            strategy_id (int): The strategy id.
+            txt_all_successful_txs (list): A list of successful transactions.
+            
+        Returns:
+            dict: The transaction data for the given strategy_id.
+        """
+        for tx in txt_all_successful_txs:
+            if str(strategy_id) in tx:
+                return json.loads(
+                    tx.split(
+                        """
+
+"""
+                    )[-1]
+                )
+
+    @staticmethod
+    def load_json_file(file_name: str, args: argparse.Namespace) -> dict:
+        """
+        This method loads a json file and returns the data as a dictionary.
+        
+        Args:
+            file_name (str): The name of the file.
+            args (argparse.Namespace): The command-line arguments.
+            
+        Returns:
+            dict: The data from the json file.
+        """
+        file_path = (
+            os.path.normpath(f"{TEST_FILE_DATA_DIR}/{file_name}")
+            if "/" not in file_name
```

Comment:
Each one of the two lines above looks OS-dependent (specifically, not compatible with Windows OS).

Reply:
This is not the case as the procedure has been tested on unix (Mac) and windows systems, and they both run as expected - the os.normpath is what normalizes the path so that it's os independent